crates/ruff_linter/src/rules/pyflakes/rules/undefined_name.rs

```diff
@@ -19,17 +19,36 @@ use ruff_macros::{derive_message_formats, violation};
 ///     return n * 2
 /// ```
 ///
+/// ## Options
+/// - [`target-version`]: Can be used to configure which symbols Ruff will understand
+///   as being available in the `builtins` namespace.
+///
 /// ## References
 /// - [Python documentation: Naming and binding](https://docs.python.org/3/reference/executionmodel.html#naming-and-binding)
 #[violation]
 pub struct UndefinedName {
     pub(crate) name: String,
+    pub(crate) minor_version_builtin_added: Option<u8>,
 }
 
 impl Violation for UndefinedName {
     #[derive_message_formats]
     fn message(&self) -> String {
-        let UndefinedName { name } = self;
-        format!("Undefined name `{name}`")
+        let UndefinedName {
+            name,
+            minor_version_builtin_added,
+        } = self;
+        let tip = minor_version_builtin_added.map(|version_added| {
+            format!(
+                "Added as a builtin in Python 3.{version_added}; \
+consider specifying a newer `target-version` in your Ruff config."
```

Comment:
What do you think of instead mentioning the version inferred by ruff. Like. 

```
Added as builtin in Python 3.{v}; but the project configuration specifies 3.{y}
```

Reply:
I'm not sure... I think the most important thing to convey to the user is "how to fix the problem". For most users who have filed bug reports, the problem isn't that they're specifying the _wrong_ `target-version` -- it's that they aren't specifying a `target-version` at all (which leads us to infer our default, Python 3.8). So I think they might be confused by a message that tells them that their project configuration specifies something

Reply:
That's fair. Part of my concern is to explicitly mention `target-version` because the version can also be configured with `requires-python.